lib/connector.rb

```diff
@@ -37,37 +37,14 @@ def self.executionSocket
 
     def self.make_connections
       @@executionSocket = TCPSocket.open(HOST_NAME, Runtime.portFromEnvVariable(GAUGE_PORT_ENV))
-      @@apiSocket = TCPSocket.open(HOST_NAME, Runtime.portFromEnvVariable(API_PORT_ENV))
     end
 
     def self.message_length(socket)
       ProtocolBuffers::Varint.decode socket
     end
 
     def self.step_value text
-      stepValueRequest = Gauge::Messages::GetStepValueRequest.new(:stepText => text)
-      apiMessage = Gauge::Messages::APIMessage.new(:messageType => Gauge::Messages::APIMessage::APIMessageType::GetStepValueRequest, :stepValueRequest => stepValueRequest)
-      response = get_api_response(apiMessage)
-      if (response.messageType == Gauge::Messages::APIMessage::APIMessageType::ErrorResponse)
-        puts "[Error] Failed to load step implementation. #{response.error.error}:  \"#{text}\""
-        return ''
-      end
-      return response.stepValueResponse.stepValue.stepValue
-    end
-
-    def self.get_api_response(apiMessage)
-      apiMessage.messageId = get_unique_id
-      dataLen = apiMessage.serialize_to_string.bytesize
-      ProtocolBuffers::Varint.encode @@apiSocket, dataLen
-      apiMessage.serialize(@@apiSocket)
-
-      responseLen = message_length(@@apiSocket)
-      data = @@apiSocket.read responseLen
-      message = Gauge::Messages::APIMessage.parse(data)
-    end
-
-    def self.get_unique_id
-      rand(2**63-1)
+      return text.gsub(/(<.*?>)/, "{}")
```

Comment:
Any reason for this change?

Reply:
We used to make an API call from runner to gauge to get stepValue of a stepText. for that, we end up having two socket connections.  As per discussion with @sriv  and @Apoorva-GA we found that there was no specific reason for that. It was just a legacy which came form gauge-java.  So we decided to remove it.